lib/omnibus/software.rb

```diff
@@ -1077,15 +1080,26 @@ def path_key
     # Apply overrides in the @overrides hash that mask instance variables
     # that are set by parsing the DSL
     #
-    def apply_overrides(attr)
+    def apply_overrides(attr, override=overrides[attr])
       val = instance_variable_get(:"@#{attr}")
-      if val.is_a?(Hash) || overrides[attr].is_a?(Hash)
+      if val.is_a?(Hash) || override.is_a?(Hash)
         val ||= {}
-        override = overrides[attr] || {}
+        override ||= {}
         val.merge(override)
       else
-        overrides[attr] || val
+        override || val
+      end
+    end
+
+    #
+    # Transform github -> git in source
+    #
+    def canonicalize_source(source)
+      if source.is_a?(Hash) && source[:github]
+        source[:git] = "https://github.com/#{source[:github]}.git"
```

Comment:
It might be worth noting (somewhere) that github URLs will use the git-over-https and not git+ssh, so users will need to have their keychain setup correctly to pull.